src/Goteo/Application/Config.php

```diff
@@ -423,36 +399,46 @@ static public function setConstants() {
             $google = [];
         }
         self::set('analytics.google', $google);
-
 	}
 
-	/**
-	 * Return a value
-	 * @param  string $name ex: filesystem.handler
-	 *                          filesystem.bucket      => array
-	 *                          filesystem.bucket.mail => string
-	 * @param  string $strick throws a Exception on fail
-	 * @return [type]       [description]
-	 */
-	static public function get($name, $strict = false) {
-		$part = strtok($name, '.');
-		if (self::$config && array_key_exists($part, self::$config)) {
-			$ret = self::$config[$part];
-			while ($part = strtok('.')) {
-				if (is_array($ret) && array_key_exists($part, $ret)) {
-					$ret = $ret[$part];
-					// echo "[$part]";
-				} elseif ($strict) {
-					throw new ConfigException("Config var [$name] not found!");
-				} else {
-					$ret = null;
-				}
-			}
-			return $ret;
-		} elseif ($strict) {
-			throw new ConfigException("Config var [$name] not found!");
-		}
-		return null;
+    /**
+     * Return a value
+     * @param string $name ex: filesystem.handler
+     *                          filesystem.bucket      => array
+     *                          filesystem.bucket.mail => string
+     * @param bool $strict
+     * @return array|false|mixed|string|null
+     * @throws ConfigException
+     */
+	static public function get(
+	    string $name,
+        bool $strict = false
+    ) {
+        $part = strtok($name, '.');
+        if (self::$config && array_key_exists($part, self::$config)) {
+            $paramValue = self::$config[$part];
+            while ($part = strtok('.')) {
+                if (is_array($paramValue) && array_key_exists($part, $paramValue)) {
+                    $paramValue = $paramValue[$part];
+                } elseif ($strict) {
+                    throw new ConfigException("Config var [$name] not found!");
+                } else {
+                    $paramValue = null;
+                }
+            }
+
+            if (!is_array($paramValue) && preg_match(self::ENV_PARAMETER_REG_EX, $paramValue, $matches)) {
+                if (sizeof($matches) >= 1) {
+                    $paramValue = getenv($matches[1]);
+                }
+            }
```

Comment:
@microstudi I've finally approached this feature using the `Config` class.

I tried refactoring `YamlSettingsLoader` , but after some time trying to apply the read of ENV parameters there, looked to me as changing too much that class.

Keeping the parsing of ENV parameters in `Config` will allow us having any kind of config file, not just Yaml as well.

To test this, you can simply do this:
1. Edit the `local-docker-settings.yml` to include an ENV variable. E.g.: `db.port: '%env(DATABASE_PORT)%'`
2. Edit the `docker-compose.yml` to have that ENV variable:
```
environment:
- DATABASE_PORT=3306
```

You'd have to run `docker-compose down` and `docker/build` in order to get the ENV variables loaded. I tried by editing the `.env` file instead of using Docker, but it just did't work :(